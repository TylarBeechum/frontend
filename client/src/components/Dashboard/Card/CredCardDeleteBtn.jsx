import React, { useState } from 'react';
import { useStateValue } from 'react-conflux';
import styled from 'styled-components';
import { Trash } from 'grommet-icons';
import { toast } from 'react-toastify';
import { BaseButton } from '../../../styles/themes';

import ConfirmationLayer from '../../ConfirmationLayer';

import queries from '../SchoolDashboard/queries';
import {
  schoolContext,
  REMOVE_CREDENTIAL_START,
  REMOVE_CREDENTIAL_SUCCESS,
  REMOVE_CREDENTIAL_ERROR
} from '../../../store/reducers/schoolReducer';

const CredCardDeleteBtn = ({ credId, credHash }) => {
  const [, dispatch] = useStateValue(schoolContext);

  const [
    hasActiveConfirmationDialog,
    setHasActiveConfirmationDialog
  ] = useState(false);

  //  Handling of loading states can be done here as well
  const confirmRemoveCredential = async e => {
    try {
      dispatch({ type: REMOVE_CREDENTIAL_START });
      await queries.removeCredential(credId, credHash);
      toast.dismiss(1);
      toast.success(`Success! Credential deleted`, {
        className: 'status-ok',
        position: toast.POSITION.BOTTOM_CENTER,
        hideProgressBar: true,
        autoClose: true
      });
      dispatch({ type: REMOVE_CREDENTIAL_SUCCESS, payload: { credId } });
    } catch {
      dispatch({ type: REMOVE_CREDENTIAL_ERROR });
    }
  };

  return (
    <>
      {/* Render the confirmation layer if it is set to be active in this component's local state */}
      {hasActiveConfirmationDialog && (
        // yesFunc for when the "Yes" button is clicked; noFunc for when the "No" button is clicked (both are optional)
        <ConfirmationLayer
          onClose={() => setHasActiveConfirmationDialog(false)} // Needed to make the layer disappear
          yesFunc={confirmRemoveCredential}
        />
      )}

      <CredCardDelBtnContainer>
        <CredCardDeleteButton
          onClick={() => setHasActiveConfirmationDialog(true)} // This state value setting will cause the layer to appear
        >
          <TrashButton />
        </CredCardDeleteButton>
      </CredCardDelBtnContainer>
    </>
  );
};

const CredCardDelBtnContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 3%;
  transition: opacity 0.3s;

  :hover {
    opacity: 0.6;
  }
`;

const TrashButton = styled(Trash)`
  cursor: pointer;
`;

const CredCardDeleteButton = styled(BaseButton)``;

export default CredCardDeleteBtn;
