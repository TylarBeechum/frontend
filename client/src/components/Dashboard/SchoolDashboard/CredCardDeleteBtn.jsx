import React from 'react'

const CredCardDeleteBtn = () => {
  return (
    <div>
        <button>delete</button>
      </div>
  )
}

export default CredCardDeleteBtn
