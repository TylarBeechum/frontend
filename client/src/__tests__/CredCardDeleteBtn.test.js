/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent } from '@testing-library/react';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../testHelpers';

import CredCardDeleteBtn from '../components/Dashboard/Card/CredCardDeleteBtn';

describe('<CredCardDeleteBtn />', () => {
  it('should successfully render the component without crashing', () => {
    const helpers = renderWithRouterAndProviders(
      <CredCardDeleteBtn credId="1" credHash="testing" />
    );
  });

  it('should match the previous snapshot of the component', () => {
    const tree = rendererWithRouterAndProviders(
      <CredCardDeleteBtn credId="5" credHash="testing" />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
