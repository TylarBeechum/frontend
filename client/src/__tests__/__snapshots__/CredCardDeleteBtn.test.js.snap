// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CredCardDeleteBtn /> should match the previous snapshot of the component 1`] = `
<div
  className="sc-EHOje gqzIeu"
>
  <button
    className="sc-bdVaJa sc-gzVnrw ibZhkv StyledButton-sc-323bzc-0 jfbKsv"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    <svg
      aria-label="Trash"
      className="sc-bZQynM IAXmQ StyledIcon-ofa7kd-0 iOkQrb"
      viewBox="0 0 24 24"
    >
      <path
        d="M4,5 L20,5 L20,23 L4,23 L4,5 Z M1,5 L23,5 M9,1 L15,1 L15,5 L9,5 L9,1 Z M9,1 L15,1 L15,5 L9,5 L9,1 Z M15,9 L15,19 M9,9 L9,19"
        fill="none"
        stroke="#000"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;
